salt/utils/network.py

```diff
@@ -2241,17 +2241,18 @@ def _test_addrs(addrinfo, port):
         if ip_addr in ip_addrs:
             continue
         ip_addrs.append(ip_addr)
-
+        s = None
         try:
             s = socket.socket(ip_family, socket.SOCK_STREAM)
             s.settimeout(2)
             s.connect((ip_addr, port))
-            s.close()
-
             ip_addrs = [ip_addr]
             break
         except OSError:
             pass
```

Comment:
Should we log the OSError?

Reply:
I don't think so. The os error is expected.